src/main/java/pl/javastart/task/CustomList.java

```diff
@@ -2,5 +2,99 @@
 
 // Uzupełnij klasę. Możesz ją dowolnie modyfikować.
 // Celem jest, żeby przechodziły testy w src/test/java/[...]/CustomListTest
-public class CustomList {
+
+import java.util.Arrays;
+
+public class CustomList<T> {
+    private Object[] elements;
+    public static final int DEFAULT_SIZE = 1;
+    private int size = 0;
+
+    public CustomList() {
+        elements = new Object[DEFAULT_SIZE];
+    }
+
+    public CustomList(int size) {
+        if (size > 0) {
+            this.elements = new Object[size];
+        } else if (size == 0) {
+            this.elements = new Object[]{};
+        } else {
+            throw new IllegalArgumentException("Illegal size: " + size);
+        }
+    }
+
+    public void add(T element) {
+        if (size == elements.length) {
+            elements = Arrays.copyOf(elements, elements.length + 1);
```

Comment:
Lepiej podwajać rozmiar niż zwiększać za każdym razem tylko o 1. Jeśli np. będziesz w pętli dodawał 1000 elementów, to za każdym razem będzie kopiowana cała dotychczasowa tablica, co jest bardzo kosztowne czasowo.